src/Console/Command/StageCommand.php

```diff
@@ -3,21 +3,72 @@
 namespace PhpTuf\ComposerStager\Console\Command;
 
 use PhpTuf\ComposerStager\Console\Misc\ExitCode;
+use PhpTuf\ComposerStager\Domain\Stager;
 use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 
 class StageCommand extends Command
 {
     protected static $defaultName = 'stage';
 
+    /**
+     * @var \PhpTuf\ComposerStager\Domain\Stager
+     */
+    private $stager;
+
+    public function __construct(Stager $stager)
+    {
+        parent::__construct(static::$defaultName);
+        $this->stager = $stager;
+    }
+
+    /**
+     * @throws \Symfony\Component\Console\Exception\InvalidArgumentException
+     */
     protected function configure(): void
     {
-        $this->setDescription('Stages a Composer command');
+        $this
+            ->setDescription('Stages a Composer command')
+            ->addArgument(
+                'composer-command',
+                InputArgument::IS_ARRAY | InputArgument::REQUIRED,
+                'The raw Composer command to stage. This MUST be preceded by a double-hyphen (" -- ") to prevent confusion of command options. See "Usage"'
+            )
+            ->addUsage('[options] -- <composer-command>...')
+            ->addUsage('-- update --with-all-dependencies')
+            ->addUsage('-- require lorem/ipsum:"^1 || ^2"')
+            ->addUsage('-- --help')
+            ->addUsage('--ansi -- help --ansi')
+            ->setHelp('If you are getting unexpected behavior from command options, be sure you are preceding the "composer-command" argument with a double-hyphen (" -- "). See "Usage"')
+        ;
     }
 
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @return int The exit code.
+     */
     public function execute(InputInterface $input, OutputInterface $output): int
     {
-        return ExitCode::SUCCESS;
+        try {
```

Comment:
Is there any chance that the scope of this `try/catch` structure can be tightened? Can we reduce the number of lines in the `try` section, and catch a more specific exception?

Reply:
I'm actually scoping it to capture absolutely any exception on purpose, because I don't want any unhandled exceptions to "blow up", resulting in difficult to interpret output. This way I can control their output exactly how I want--which incidentally is just printing whatever their `$message` is but with consistent formatting and exit code with the rest of the command's errors conditions.